commit f8da5752fd1b25f1ecf78a79013e2dfd2b860589
Author: Marc Zyngier <maz@kernel.org>
Date:   Mon Feb 8 09:57:31 2021 +0000

    arm64: cpufeatures: Allow disabling of Pointer Auth from the command-line
    
    In order to be able to disable Pointer Authentication  at runtime,
    whether it is for testing purposes, or to work around HW issues,
    let's add support for overriding the ID_AA64ISAR1_EL1.{GPI,GPA,API,APA}
    fields.
    
    This is further mapped on the arm64.nopauth command-line alias.
    
    Signed-off-by: Marc Zyngier <maz@kernel.org>
    Reviewed-by: Catalin Marinas <catalin.marinas@arm.com>
    Acked-by: David Brazdil <dbrazdil@google.com>
    Tested-by: Srinivas Ramana <sramana@codeaurora.org>
    Link: https://lore.kernel.org/r/20210208095732.3267263-23-maz@kernel.org
    Signed-off-by: Will Deacon <will@kernel.org>

diff --git a/arch/arm64/include/asm/cpufeature.h b/arch/arm64/include/asm/cpufeature.h
index 30917b9a760b..61177bac49fa 100644
--- a/arch/arm64/include/asm/cpufeature.h
+++ b/arch/arm64/include/asm/cpufeature.h
@@ -820,6 +820,7 @@ static inline unsigned int get_vmid_bits(u64 mmfr1)
 
 extern struct arm64_ftr_override id_aa64mmfr1_override;
 extern struct arm64_ftr_override id_aa64pfr1_override;
+extern struct arm64_ftr_override id_aa64isar1_override;
 
 u32 get_kvm_ipa_limit(void);
 void dump_cpu_features(void);